src/Arcus.Observability.Tests.Integration/Serilog/Sinks/ApplicationInsights/AzureKeyVaultDependencyTests.cs

```diff
@@ -48,12 +53,22 @@ public async Task LogAzureKeyVaultDependency_SinksToApplicationInsights_ResultsI
                     Assert.NotEmpty(results.Value);
                     Assert.Contains(results.Value, result =>
                     {
-                        return result.Dependency.Type == "Azure key vault"
+                        return result.Dependency.Type == dependencyType
                                && result.Dependency.Target == vaultUri
                                && result.Dependency.Data == secretName;
                     });
                 });
             }
+
+            Assert.Contains(GetLogEventsFromMemory(), logEvent =>
+            {
+                StructureValue logEntry = logEvent.Properties.GetAsStructureValue(ContextProperties.DependencyTracking.DependencyLogEntry);
+                return logEntry != null
+                       && logEntry.Properties.FirstOrDefault(prop => prop.Name == nameof(DependencyLogEntry.DependencyType))?.Value.ToDecentString() == dependencyType
+                       && logEntry.Properties.FirstOrDefault(prop => prop.Name == nameof(DependencyLogEntry.DependencyData))?.Value.ToDecentString() == secretName
+                       && logEntry.Properties.FirstOrDefault(prop => prop.Name == nameof(DependencyLogEntry.TargetName))?.Value.ToDecentString() == vaultUri
+                       && logEntry.Properties.FirstOrDefault(prop => prop.Name == nameof(DependencyLogEntry.Context)) != null;
+            });
```

Comment:
Maybe better to use our upgraded `AssertX.Any` for this as a failure message from this kind of assertion will be rather obscure.

Note that this suggestion is not complete. 😄 + suggestion for all similar assertions.
```suggestion
            AssertX.Any(GetLogEventsFromMemory(), logEvent =>
            {
                StructureValue logEntry = Assert.Single(logEvent.Properties, property => property.GetAsStructureValue(ContextProperties.DependencyTracking.DependencyLogEntry));
                Assert.NotNull(logEntry);
                
                var actualDependencyType = Assert.Single(logEntry.Properties, prop => prop.Name == nameof(DependencyLogEntry.DependencyType));
                Assert.Equal(dependencyType, actualDependencyType);
            });
```

Reply:
Updated the pull request to use AssertX for assertion.